nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -319,6 +319,38 @@ in
         };
       };
 
+      defaultSession = mkOption {
+        type = types.str;
+        default = let
+            dmDefault = cfg.desktopManager.default;
+            wmDefault = cfg.windowManager.default;
+            defaultPackage =
+              if cfg.displayManager.sessionPackages != [] then
+                (head (head cfg.displayManager.sessionPackages).providedSessions)
+              else
+                null;
+            defaultDmWm = dmDefault + optionalString (wmDefault != "none") ("+" + wmDefault);
+          in
+            if defaultDmWm == "none" && isString defaultPackage then
+              defaultPackage
+            else
+              defaultDmWm;
+        example = "gnome";
+        description = "Default graphical session (only effective for LightDM and SDDM).";
+        apply = d: let
+            sessionNames = cfg.displayManager.session.names ++
+              (concatMap (p: p.providedSessions) cfg.displayManager.sessionPackages);
```

Comment:
I'm abusing `apply` here a bit to get a sane error message. Could move it to the type check, but would still need an explicit throw to get an error which actually tells the user what's wrong.

Reply:
Could we use `lib.assertMsg`?

Reply:
That would work.